### ISSUE TITLE
关于多输入标签的问题

### ISSUE THREAD

**Issue by lxk767363331** [PaddlePaddle/PaddleSeg#339]
我想在paddleseg中输入2个label，尝试使用合并label输入失败了，想问一下在哪里怎么修改可以输入2个label

**Comment by wuyefeilin** [comment]
请问下，你是在什么场景下使用的，能具体描述一下吗？ 

你说的合并label是怎么一个合并法，两个label有什么不同？

**Comment by lxk767363331** [comment]
你好，我是在复现这个文章：Gated-SCNN: Gated Shape CNNs for Semantic Segmentation的时候发现需要3个损失函数，其中一个损失函数需要的是边缘信息，我生成了边缘的label 但是损失函数上会出现问题，所以想问一下如何设置 输入2个label

**Comment by wuyefeilin** [comment]
如果你的边缘label可以从常规label生成的话，直接在计算loss的时候调用算子进行处理就行了。

如果你的边缘label是单独的标注图像的话，得在下面这些地方进行处理：
1. https://github.com/PaddlePaddle/PaddleSeg/blob/release/v0.5.0/pdseg/models/model_builder.py#L133 得在这边增加label输入接口， 然后在loss模块得作相应的处理
2. https://github.com/PaddlePaddle/PaddleSeg/blob/release/v0.5.0/pdseg/reader.py#L248 在这里得增加对边缘label的返回
3. https://github.com/PaddlePaddle/PaddleSeg/blob/release/v0.5.0/pdseg/train.py#L198 这边这个函数也要作出相应修改
4. 然后就是数据增强这一块得对边缘label像常规label一样进行处理。

**Comment by lxk767363331** [comment]
问题1：边缘label是在原本的label产生的，就是使用1个像素的大小表示。我在torch里面找到了对应的库做到了，不知道在paddle里面使用哪方面的
![image](https://user-images.githubusercontent.com/38516795/89871497-163abd80-dbea-11ea-9900-32778b43ae70.png)

![image](https://user-images.githubusercontent.com/38516795/89871476-0a4efb80-dbea-11ea-8db4-dfb557c5d863.png)

问题2：对于这个边缘label 白色边缘是255 正常的label好像不可以设置为255 那我是不是需要改成别的数值呢

**Comment by wuyefeilin** [comment]
1. 请问一下torch是哪个库可以实现这个功能的，我这边对照一下paddle有没有类似的api

2. 我们提供的loss会对255进行忽略， 边缘分类是一个二分类的问题，你应该把数值改成1

**Comment by lxk767363331** [comment]
https://github.com/nv-tlabs/GSCNN/blob/3648b86d822f3e73ab1d8e827cf47466bf8f01a7/datasets/cityscapes.py#L219
https://github.com/nv-tlabs/GSCNN/blob/3648b86d822f3e73ab1d8e827cf47466bf8f01a7/datasets/cityscapes.py#L231
这里是他生成的地方 我是直接拿过来生成的然后放到了数据集里面

**Comment by lxk767363331** [comment]
import transforms.transforms as extended_transforms
from PIL import Image
import torch
from torchvision import transforms
import numpy as np
from scipy.ndimage.morphology import distance_transform_edt
path="Blowhole0.png"
spath="Blowhole_edge.png"
def mask_to_onehot(mask, num_classes):
    _mask = [mask == (i + 1) for i in range(num_classes)]
    return np.array(_mask).astype(np.uint8)

def onehot_to_binary_edges(mask, radius, num_classes):
    if radius < 0:
        return mask
    # We need to pad the borders for boundary conditions
    mask_pad = np.pad(mask, ((0, 0), (1, 1), (1, 1)), mode='constant', constant_values=0)
    edgemap = np.zeros(mask.shape[1:])
    for i in range(num_classes):
        dist = distance_transform_edt(mask_pad[i, :]) + distance_transform_edt(1.0 - mask_pad[i, :])
        dist = dist[1:-1, 1:-1]
        dist[dist > radius] = 0
        edgemap += dist
    edgemap = np.expand_dims(edgemap, axis=0)
    edgemap = (edgemap > 0).astype(np.uint8)
    return edgemap

mask = Image.open(path)
mask = torch.from_numpy(np.array(mask, dtype=np.int32)).long()
_mask = mask.numpy()
_edgemap = mask_to_onehot(_mask, 3)
_edgemap = onehot_to_binary_edges(_edgemap, 1, 3)
edgemap = torch.from_numpy(_edgemap).float()
unloader = transforms.ToPILImage()
image = edgemap.cpu().clone()  # clone the tensor
image = image.squeeze(0)  # remove the fake batch dimension
image = unloader(image)
image.save(spath)

这是我转换的代码
